src/snowflake/snowpark/_internal/analyzer/config_context.py

```diff
@@ -0,0 +1,37 @@
+#
+# Copyright (c) 2012-2024 Snowflake Computing Inc. All rights reserved.
+#
+
+from typing import Any
+
+
+class ConfigContext:
+    """Class to help reading a snapshot of configuration attributes from a session object.
+
+    On instantiation, this object stores the configuration from the session object
+    and returns the stored configuration attributes when requested.
+    """
+
+    def __init__(self, session) -> None:
+        self.session = session
+        self.configs = {
+            "_query_compilation_stage_enabled",
+            "cte_optimization_enabled",
```

Comment:
instead of doing this, it might be more clear to provide a property interface in the session that teaks a snapshot. so it is the interface is at a closer place with the properties.

Given the fact that we are adding more and more configs,  i am thinking it might worth to do a refactoring to centralize all configs, and wrap all of them into a class to reduce the possiblility for developer make mistakes.